### PR TITLE
Fix pretty query parameter for v0 API

### PR DESCRIPTION
**Pull request by clarshad**
This fixes indentation if pretty query parameter is set for v0 API
No major feature modification or changes in method, hence made no change to unit test 

Fixes: #3332
Signed-off-by: Arshad Saquib <arshad.saquib@styra.com>

### CODE REVIEW

**Comment by anderseknert** [comment]
Looks like it would be easy enough to add a test case [here](https://github.com/open-policy-agent/opa/blob/1136df0bfc401426799c21392a5040f9eb5062b2/server/server_test.go#L434
), no?

Something like:
```go
pretty := `{
    ...pretty printed body here...
}`

if err := f.v0(http.MethodPost, "/data/test?pretty=true", `{"flag": true}`, 200, pretty); err != nil {
    t.Fatalf("Expected response [1,2,3,4] but got: %v", err)
}
```

File: server/server.go [hunk 1]
```diff
@@ -844,6 +844,7 @@ func (s *Server) v0QueryPath(w http.ResponseWriter, r *http.Request, urlPath str
 	decisionID := s.generateDecisionID()
 
 	ctx := r.Context()
+	pretty := getBoolParam(r.URL, types.ParamPrettyV1, true)
```
Comment (anderseknert) [inline server/server.go:847]: The default should not be true but false.

Also this is assigned more than 100 lines away from where it's used. I'd move this down to line 954.

**Review by anderseknert — CHANGES_REQUESTED** [review]
Some suggestions. And a test would be useful.

File: server/server.go (hunk 1 above)
Comment (clarshad) [reply server/server.go:847]: @anderseknert took reference from [v1DataPost](https://github.com/open-policy-agent/opa/blob/b4d5a234d0dd284b2f822a24791afd2d49756a42/server/server.go#L1364)
Seems like default is set to true wherever  `getBoolParam` is called, not sure why
Also in `v1DataPost` method, retrieving all details from `r http.Request` is done initially and then later utilised at different places, hence was thinking to have it consistent for v0 and v1 
However logically it does make sense to do both - set default as false and move assignment down the line

**Comment by clarshad** [comment]
> Looks like it would be easy enough to add a test case [here](https://github.com/open-policy-agent/opa/blob/1136df0bfc401426799c21392a5040f9eb5062b2/server/server_test.go#L434), no?
> 
> Something like:
> 
> ```go
> pretty := `{
>     ...pretty printed body here...
> }`
> 
> if err := f.v0(http.MethodPost, "/data/test?pretty=true", `{"flag": true}`, 200, pretty); err != nil {
>     t.Fatalf("Expected response [1,2,3,4] but got: %v", err)
> }
> ```

make sense ! Will add it

File: server/server.go (hunk 1 above)
Comment (anderseknert) [reply server/server.go:847]: Oh! I was wrong - it was not a default but "ifEmpty".. so if the parameter is provided _but_ without a value (like `/v0/data/test?pretty`) it should be counted as true.. so please leave it as it was then - it should still be moved though.

Comment (clarshad) [reply server/server.go]: done

**Review by anderseknert — APPROVED** [review]

**Comment by anderseknert** [comment]
Thanks @clarshad ! 👍

**Pull request merged**